### PR TITLE
Enh control tick deconflict2

### PR DESCRIPTION
**Pull request by tacaswell**
This is an alternative to #13826 that puts the knob on the `Axis` rather than the locator.

I have a memory of talking about doing this and deciding not to, but do not remember _why_.

attn @efiring

### CODE REVIEW

File: lib/matplotlib/axis.py [hunk 1]
```diff
@@ -754,6 +756,17 @@ def __init__(self, axes, pickradius=15):
     majorTicks = _LazyTickList(major=True)
     minorTicks = _LazyTickList(major=False)
 
+    def get_remove_overlapping_locs(self):
```
Comment (anntzer) [inline lib/matplotlib/axis.py]: I thought we decided not to bother with getter and setters?

**Comment by anntzer** [comment]
I indeed missed the fact that this now lives on an Artist subclass and therefore the setter is useful to play nicely with Artist.set, so I'm fine with keeping it.

**Comment by jklymak** [comment]
This looks fine to me.  Definitely needs documentation somewhere...

**Comment by tacaswell** [comment]
API changes notes are pushed to #13870 I don't think this needs a more prominent tutorial or example.

**Comment by tacaswell** [comment]
(specifically https://github.com/matplotlib/matplotlib/pull/13870/commits/6cbbc854410358ac7301df923823f9824d8d1c31 )

**Comment by tacaswell** [comment]
Not clear what is wrong with appeyor...

**Comment by anntzer** [comment]
rebase and push?

**Comment by tacaswell** [comment]
rebased on current master and force-pushed, appveyor looks happier (it started!).

**Comment by jklymak** [comment]
> API changes notes are pushed to #13870 I don't think this needs a more prominent tutorial or example.

OK, but so far as I can see this has *no* documentation at all except for the API note?  https://20198-1385122-gh.circle-artifacts.com/0/home/circleci/project/doc/build/html/api/axis_api.html#axis-objects

**Comment by tacaswell** [comment]
Pushed a docs change.

**Pull request merged**